pkg/util/lora/lora.go

```diff
@@ -27,3 +33,38 @@ func AdjustedRSSI(channelRSSI, snr float32) float32 {
 	}
 	return rssi
 }
+
+// ExtractRxMetadataReceivedAt returns the ReceivedAt timestamp based on the
+// provided metadatas or nil if no md.GpsTime nor md.ReceivedAt is found.
+func ExtractRxMetadataReceivedAt(mds []*ttnpb.RxMetadata) *pbtypes.Timestamp {
```

Comment:
You can merge these 3 functions into one if you accept an interface with the following contents:
```
GetRxMetadata() []*RxMetadata
GetReceivedAt() *timestamppb.Timestamp
```